bindata/assets/alerts/client-go-requests.yaml

```
@@ -0,0 +1,53 @@
+apiVersion: monitoring.coreos.com/v1
+kind: PrometheusRule
+metadata:
+  name: client-go
+  namespace: openshift-kube-apiserver
+spec:
+  groups:
+    - name: control-plane-client-go
+      rules:
+        # latency by:
+        # - job (kubelet, apiserver. controller-manager, ...)
+        # - destination host: internal-lb, kubernetes.default service, direct connection ...
+        - record: cluster:rest_client_request_duration_seconds:internal_load_balancer:histogram_quantile
+          labels:
+            quantile: "0.99"
+          expr: |
+            histogram_quantile(0.99,
+              sum(rate(
+                label_replace(rest_client_request_duration_seconds_bucket{verb="GET",url=~"https?://api-int.*"},
+                  "host","$1","url","https?://([^/\\s]+).*")[5m:30s]
+                )
+              ) by (le,host,service,namespace,node)
+            )
+        - record: cluster:rest_client_request_duration_seconds:services:histogram_quantile
+          labels:
+            quantile: "0.99"
+          expr: |
+            histogram_quantile(0.99,
+              sum(rate(
+                label_replace(rest_client_request_duration_seconds_bucket{verb="GET",url!~"https?://(api-int|\\[::1\\]|127\\.0\\.0\\.1|localhost).*"},
+                  "host","$1","url","https?://([^/\\s]+).*")[5m:30s]
+                ) * on (cluster_ip) group_left group(kube_service_info{cluster_ip!~"None",namespace=~"(openshift|kube|default).*"}) by (service,namespace,cluster_ip)
+              ) by (le,host,service,namespace)
+            )
+        - record: cluster:rest_client_request_duration_seconds:localhost:histogram_quantile
```

Maybe `pod` is better suited for the scope wdyt?
```suggestion
        - record: pod:rest_client_request_duration_seconds:histogram_quantile
```

---

is pod really the scope?
I think is better scoped by the destination than by the origin, is more about control_plane or something like that,